### PR TITLE
Update storage

### PR DESCRIPTION
**Pull request by JoelHo**
Migrate Pool passengers Set -> List
Add pools read from JSON to AddressBook

### CODE REVIEW

File: src/main/java/seedu/address/model/pool/Pool.java [hunk 1]
```diff
@@ -95,8 +112,7 @@ public boolean isSamePool(Pool otherPool) {
         return otherPool != null
                 && otherPool.getDriver().equals(getDriver())
                 && otherPool.getTripDay().equals(getTripDay())
-                && otherPool.getTripTime().equals(getTripTime())
-                && otherPool.getPassengers().equals(getPassengers());
```
Comment (lyueyang) [inline src/main/java/seedu/address/model/pool/Pool.java:99]: sorry, why's it no longer a requirement to check for tripTime?

**Review by lyueyang — APPROVED** [review]
everything LGTM apart from the little part about doing the equals, does it make sense if pools with different passengers to be equal?

File: src/main/java/seedu/address/model/pool/Pool.java (hunk 1 above)
Comment (JoelHo) [reply src/main/java/seedu/address/model/pool/Pool.java:99]: This method is used to ensure we don't create a "duplicate" pool, so if we checked for passenger equality, the same driver could potentially be driving 2 pools at the same time as long as the passengers were different.

Comment (lyueyang) [reply src/main/java/seedu/address/model/pool/Pool.java:99]: right, that makes sense, thanks!

**Review by ZechariahTan — APPROVED** [review]
LGTM

**Pull request merged**